datafusion/core/src/datasource/physical_plan/parquet/schema_adapter.rs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow_array::RecordBatch;
+use arrow_schema::{Schema, SchemaRef};
+use std::fmt::Debug;
+use std::sync::Arc;
+
+/// Factory of schema adapters.
```

Comment:
I know that currently the only user of SchemaAdapter is parquet, but I don't think there is anything parquet specific about the logic here. 

What do you think about moving the code (and default impl) somewhere like 

```
datafusion/core/src/datasource/schema_adapter.rs
```

? 

Perhaps we could do that as a follow on PR as the way you have done this PR makes it easy to see what you have changed / not changed 👍